llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -6629,6 +6629,11 @@ void SITargetLowering::ReplaceNodeResults(SDNode *N,
       Results.push_back(LoadVal);
       return;
     }
+    case Intrinsic::amdgcn_dead: {
```

Comment:
Why do you need to handle this both here in `ReplaceNodeResults` and in `LowerINTRINSIC_WO_CHAIN` below?